src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ILEmit/ILEmitResolverBuilder.cs

```diff
@@ -100,8 +100,10 @@ private GeneratedMethod BuildTypeNoCache(ServiceCallSite callSite)
                 owner: GetType(),
                 skipVisibility: true);
 
-            ILEmitCallSiteAnalysisResult info = ILEmitCallSiteAnalyzer.Instance.CollectGenerationInfo(callSite);
-            ILGenerator ilGenerator = dynamicMethod.GetILGenerator(info.Size);
+            // In traces we've seen methods range from 100B - 4K sized methods since we've
+            // stop trying to inline everything into scoped methods. We'll pay for a couple of resizes
+            // so there'll be allocations but we could potentially change ILGenerator to use the array pool
+            ILGenerator ilGenerator = dynamicMethod.GetILGenerator(512);
```

Comment:
Is it worth moving this magic number out into a const and adding a further remark of why 512 was chosen?

Reply:
I would normally agree if it used in more than on place. I could add size: 512 to make it a bit clearer but the comment is inline.